webapp/channels/src/components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -36,508 +63,301 @@ type Props = PropsFromRedux & {
-            channelNotifyProps?.push === pushNotifyLevel &&
-            channelNotifyProps?.push_threads === pushThreadsNotifyLevel
-        ) {
-            this.updateSection(NotificationSections.NONE);
-            return;
-        }
-
-        const props = {push: pushNotifyLevel, push_threads: pushThreadsNotifyLevel};
-        this.handleUpdateChannelNotifyProps(props);
-    };
-
-    handleUpdatePushNotificationLevel = (pushNotifyLevel: ChannelNotifyProps['push']) => this.setState({pushNotifyLevel});
-    handleUpdatePushThreadsNotificationLevel = (pushThreadsNotifyLevel: UserNotifyProps['push_threads']) => this.setState({pushThreadsNotifyLevel});
-    handleUpdateIgnoreChannelMentions = (ignoreChannelMentions: ChannelNotifyProps['ignore_channel_mentions']) => this.setState({ignoreChannelMentions});
-
-    handleSubmitIgnoreChannelMentions = () => {
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props;
-        const {ignoreChannelMentions} = this.state;
-
-        if (channelNotifyProps?.ignore_channel_mentions === ignoreChannelMentions) {
-            this.updateSection(NotificationSections.NONE);
-            return;
-        }
-
-        const props = {ignore_channel_mentions: ignoreChannelMentions};
-        this.handleUpdateChannelNotifyProps(props);
-    };
-
-    handleUpdateChannelAutoFollowThreads = (channelAutoFollowThreads: ChannelNotifyProps['channel_auto_follow_threads']) => this.setState({channelAutoFollowThreads});
-
-    handleSubmitChannelAutoFollowThreads = () => {
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props;
-        const {channelAutoFollowThreads} = this.state;
-
-        if (channelNotifyProps?.channel_auto_follow_threads === channelAutoFollowThreads) {
-            this.updateSection(NotificationSections.NONE);
-            return;
-        }
-
-        const props = {channel_auto_follow_threads: channelAutoFollowThreads};
-        this.handleUpdateChannelNotifyProps(props);
-    };
-
-    render() {
-        const {
-            activeSection,
-            desktopNotifyLevel,
-            desktopThreadsNotifyLevel,
-            desktopSound,
-            desktopNotifySound,
-            markUnreadNotifyLevel,
-            pushNotifyLevel,
-            pushThreadsNotifyLevel,
-            ignoreChannelMentions,
-            channelAutoFollowThreads,
-            serverError,
-        } = this.state;
-
-        const {
-            channel,
-            channelMember,
-            currentUser,
-            sendPushNotifications,
-        } = this.props;
-
-        const isNotificationsSettingSameAsGlobal = this.verifyNotificationsSettingSameAsGlobal({
-            desktopNotifyLevel,
-            desktopNotifySound,
-            desktopSound,
-            desktopThreadsNotifyLevel,
+        await props.actions.updateChannelNotifyProps(props.currentUser.id, props.channel.id, userSettings).then((value) => {
```

Comment:
Unless we're `await`ing the `handleSave`, I think you can remove the `async` on that and the `await` here. It doesn't seem like those should be needed since we're using `.then` to handle the result of the promise

Reply:
Should there also be some sort of indication that this modal is saving? I don't see anything in the design for that

CC @asaadmahmood